homeworks/rostyslav.havuka_RostyslavHavuka/6-typeScript/src/index.ts

```diff
@@ -0,0 +1,247 @@
+const scoreElement = document.querySelector('[data-score]') as HTMLHeadingElement;
+const cubeElement = document.querySelector('[data-cube]') as HTMLDivElement;
+const cubeScoreElement = document.querySelector('[data-cube-score]') as HTMLDivElement;
+const keyElement = document.querySelector('[data-key]') as HTMLInputElement;
+const progressBarElement = document.querySelector('[data-progress-bar]') as HTMLDivElement;
+const keyStart = document.querySelector('[data-key-start]') as HTMLButtonElement
+const keyStop = document.querySelector('[data-key-stop]') as HTMLButtonElement
+const keyAgain = document.querySelector('[data-key-again]') as HTMLButtonElement
+
+class Game {
+  private currentKey: string = '';
+  private score: number = 100;
+  private interval: number = 2000;
+  private intervalId: NodeJS.Timeout;
+  private intervalStatusBarId: NodeJS.Timeout;
+  private gameState: boolean = false;
+  private timerProgressBar: number = 0;
+  private width: number;
+
+  constructor(
+    private scoreElement: HTMLHeadingElement,
+    private cubeScoreElement: HTMLDivElement,
+    private keyElement: HTMLInputElement,
+    private cubeElement?: HTMLDivElement,
+    private progressBarElement?: HTMLDivElement,
+  ) { }
+
+  start() {
+    window.addEventListener('load', this.startKeysInterval())
```

Comment:
There are several points I have:

1. This function (`this.startKeysInterval`) will be called immediately because you should pass there a function `this.startKeysInterval`, but no a function call `this.startKeysInterval()`.
So you need replace `window.addEventListener('load', this.startKeysInterval())` with window.addEventListener('load', this.startKeysInterval);

2. Your game should not be dependent on window events. It should work immediately when the `start` method is called. So it's better to use it just like this:

```ts
start() {
   this.startKeysInterval();
   // ...
}

// ....
const game = new Game(scoreElement, cubeScoreElement, keyElement, cubeElement, progressBarElement);

window.addEventListener('load', game.start);
```

3. But do you really need to wait for the load event? 😄 
My code worked without any waiting. Did you face some problems that forced you to use load listener?